src/server/scripts/Spells/spell_warlock.cpp

```diff
@@ -1490,6 +1494,78 @@ class spell_warl_volatile_agony : public SpellScript
     }
 };
 
+// Bilescourge Bombers - 267211
+class spell_warl_bilescourge_bombers : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_WARLOCK_BILESCOURGE_BOMBERS_AREATRIGGER });
+    }
+
+    void HandleLaunch(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetCaster()->GetPosition(), SPELL_WARLOCK_BILESCOURGE_BOMBERS_AREATRIGGER, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringSpell = GetSpell()
+            });
+    }
+
+    void Register() override
+    {
+        OnEffectLaunch += SpellEffectFn(spell_warl_bilescourge_bombers::HandleLaunch, EFFECT_0, SPELL_EFFECT_CREATE_AREATRIGGER);
+    }
+};
+
+// Bilescourge Bombers - 15141
+struct areatrigger_warl_bilescourge_bombers : AreaTriggerAI
+{
+    areatrigger_warl_bilescourge_bombers(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger), _tickTimer(TICK_PERIOD) {}
+
+    static constexpr Milliseconds TICK_PERIOD = Milliseconds(500);
+
+    void OnUpdate(uint32 diff) override
+    {
+        _tickTimer -= Milliseconds(diff);
+
+        while (_tickTimer <= 0s)
+        {
+            if (Unit* caster = at->GetCaster())
```

Comment:
i'd move the caster if outside the while. Reason being: the while-body is only executed multiple times if we have a lag, and if that happens `GetCaster()` is valid throughout the entire iteration, so we dont have to nullcheck it each iteration but only once before iterations happen